### PR TITLE
[plugins] Obfuscate proxy credentials in env. settings

### PR DESCRIPTION
**Pull request by pmoravec**
Obfuscate credentials in env.variable settings for e.g. HTTP_PROXY or similar.

Inspired by: #3788
Resolves: #3789

---
Please place an 'X' inside each '[]' to confirm you adhere to our [Contributor Guidelines](https://github.com/sosreport/sos/wiki/Contribution-Guidelines)

- [X] Is the commit message split over multiple lines and hard-wrapped at 72 characters?
- [X] Is the subject and message clear and concise?
- [X] Does the subject start with **[plugin_name]** if submitting a plugin patch or a **[section_name]** if part of the core sosreport code?
- [X] Does the commit contain a **Signed-off-by: First Lastname <email@example.com>**?
- [X] Are any related Issues or existing PRs [properly referenced](https://docs.github.com/en/issues/tracking-your-work-with-issues/creating-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) via a Closes (Issue) or Resolved (PR) line?
- [X] Are all passwords or private data gathered by this PR [obfuscated](https://github.com/sosreport/sos/wiki/How-to-Write-a-Plugin#how-to-prevent-collecting-passwords)?

### CODE REVIEW

**Comment by pmoravec** [comment]
Trivial reproducer:

    echo HTTP_PROXY=http://foouser:barpassword@proxyurl:8080 >> /etc/environment
    sos report -o system --batch --build

I am not sure if whole `/etc/sysconfig` or `/etc/default` needs such scrubbing - various services can define the proxy there so it is imho better to cover all. BUT if either directory is password-less every time, I am happy to remove it from the `postproc`.

**Comment by pmoravec** [comment]
OK, the `/etc/sysconfig` or `/etc/default` could also collect the env.variables, worth processing them as well.

File: sos/report/plugins/anaconda.py [hunk 1]
```diff
@@ -24,21 +24,21 @@ class Anaconda(Plugin, RedHatPlugin):
...
         else:
-            paths = paths + [
+            self.copypaths = self.copypaths + [
                 "/var/log/anaconda.*",
                 "/root/install.log",
                 "/root/install.log.syslog"
             ]
 
-        self.add_copy_spec(paths)
+        self.add_copy_spec(self.copypaths)
```
Comment (pmoravec) [inline sos/report/plugins/anaconda.py:41]: The credentials can be in either anaconda log or in the `*ks.cfg` file - hence apply postproc to all copyspecs.

**Comment by jcastill** [comment]
The pylint failure is solved already but checking out this pr doesn't show the commit, so I think you'll have to resync.

**Review by jcastill — APPROVED** [review]

**Review by arif-ali — APPROVED** [review]
looks good to me, it would be good to do the same in apt.py as well, but if you prefer, I can do a seperate PR for that later (as that is already obfuscating proxy)

**Comment by pmoravec** [comment]
Good point wrt Apt plugin where I in fact copied the RE from - I just updated the PR accordingly.

**Review by TurboTurtle — APPROVED** [review]
This LGTM, however I should note that what this _does_ and what it _says_ it does are slightly different.

The obfuscation isn't actually looking for `*_PROXY` variables, just urls with username and passwords in them, which _so happen_ to be the convention for proxy authentication.

I have a slight preference to rename to `do_path_http_sub()` or something similar, but it is not a big item for me and I'm fine if you want to merge as is.

**Pull request closed without merging**